### ISSUE TITLE
ARGB32 (and/or AGray32) test failure on aarch64

### ISSUE THREAD

**Issue by Millak** [JuliaGraphics/ColorTypes.jl#288]
I'm attempting to package ColorTypes.jl in Guix and I came across a test failure on aarch64. I have 25 instances of:

> RGB24/ARGB32 constructors: Test Failed at /gnu/store/ms5wncdgzv1p90xahj6pk6xk8lfx4c6a-julia-colortypes-0.11.1/share/julia/loadpath/ColorTypes/test/types.jl:177
>  Expression: ARGB32(c0, 0.8) === ARGB32(val1, val2, val1, 0.8)
>   Evaluated: ARGB32(0.2N0f8, 0.6N0f8, 0.2N0f8, 0.0N0f8) === ARGB32(0.2N0f8, 0.6N0f8, 0.2N0f8, 0.8N0f8)
>Stacktrace:
> [1] macro expansion
>   @ /gnu/store/q92hjagi3kxzqajvx1qp1jz7az747y98-julia-1.8.3/share/julia/stdlib/v1.8/Test/src/Test.jl:464 [inlined]
> [2] macro expansion
>   @ /gnu/store/ms5wncdgzv1p90xahj6pk6xk8lfx4c6a-julia-colortypes-0.11.1/share/julia/loadpath/ColorTypes/test/types.jl:177 [inlined]
> [3] macro expansion
>   @ /gnu/store/q92hjagi3kxzqajvx1qp1jz7az747y98-julia-1.8.3/share/julia/stdlib/v1.8/Test/src/Test.jl:1360 [inlined]
> [4] top-level scope
>   @ /gnu/store/ms5wncdgzv1p90xahj6pk6xk8lfx4c6a-julia-colortypes-0.11.1/share/julia/loadpath/ColorTypes/test/types.jl:170

I don't actually do julia programming so I wasn't able to try to propose a patch for this. I've attached the full build log.

[julia-colortypes-0.11.1.log](https://github.com/JuliaGraphics/ColorTypes.jl/files/10226614/julia-colortypes-0.11.1.log)

**Comment by kimikage** [comment]
I have added unit tests for aarch64 using QEMU(#301 and #303).
As for v0.11.5 on nightly build, it passed the aarch64 test.
```
  Julia Version 1.12.0-DEV.406
  Commit 8f6418ea507 (2024-04-25 04:04 UTC)
  Build Info:
    Official https://julialang.org/ release
  Platform Info:
    OS: Linux (aarch64-linux-gnu)
    CPU: 4 × unknown
    WORD_SIZE: 64
    LLVM: libLLVM-16.0.6 (ORCJIT, generic)
  Threads: 1 default, 0 interactive, 1 GC (on 4 virtual cores)
```
On the other hand, with v1.10.2 and v1.6.7, the error was reproduced.
```
  transparent gray constructors: Test Failed at /home/runner/work/ColorTypes.jl/ColorTypes.jl/test/types.jl:268
    Expression: AGray32(val, 0.8) === AGray32(0.2, 0.8)
     Evaluated: AGray32(0.2N0f8,0.0N0f8) === AGray32(0.2N0f8,0.8N0f8)
```

**Comment by kimikage** [comment]
Curiously, for ColorTypes v0.12.0-DEV, julia v1.10.2 and nightly passed, julia v1.6.7 failed.

**Comment by kimikage** [comment]
I'm guessing this is an `_unsafe_trunc` issue, but if so, it seems like a `FixedPointNumbers` issue.

**Comment by kimikage** [comment]
I am attempting to create an MWE in: https://github.com/kimikage/ColorTypes.jl/pull/3

**Comment by kimikage** [comment]
Surprisingly, the cause of this does not seem to be an ARM issue.
It is just triggered by the workaround for ARM in FixedPointNumbers.jl, and the same problem could also occur on x64.

**Comment by kimikage** [comment]
fixed by FixedPointNumbers 0.8.5
